card/mod/core/set/all/name_validations.rb

```diff
@@ -5,8 +5,8 @@
 end
 
 event :validate_uniqueness_of_codename do
-  return unless codename.present? && errors.empty? &&
-                Card.find_by_codename(codename).present?
+  return (self.codename = nil) unless codename.present?
+  return unless errors.empty? && Card.find_by_codename(codename).present?
```

Comment:
I find the negated versions easier to read, e.g.`if errors.present? || Card.find_by_codename(codename).blank?`